adminutils/adminutils.py

```diff
@@ -185,32 +185,61 @@ async def emoji_add(self, ctx, name: str, url: str, *roles: discord.Role):
             await ctx.tick()
 
-    @emoji.command(name="message", aliases=["steal"])
-    async def emote_steal(self, ctx, name: str, message_id: discord.Message, *roles: discord.Role):
+    @emoji.command(name="import")
+    async def emoji_import(
+        self,
+        ctx,
+        name: str,
+        location: Union[discord.Message, discord.Member, discord.Emoji, discord.PartialEmoji],
+        *roles: discord.Role,
+    ):
         """
-        Add an emoji from a specified message
+        Add an emoji from either a member's status, a message, or an emoji from another server.
+
         Use double quotes if role name has spaces
 
         Examples:
-            `[p]emoji message Example 162379234070467641`
-            `[p]emoji message RoleBased 162379234070467641 EmojiRole`
+            `[p]emoji import 236598364265634245`
+            `[p]emoji import Username#0000 peepoDance`
+            `[p]emoji import :thonk: thonk emojiRole`
         """
-        # TrusyJaid NotSoBot converter
-        # https://github.com/TrustyJAID/Trusty-cogs/blob/a3e931bc6227645007b37c3f4f524c9fc9859686/notsobot/converter.py#L30-L36
-        emoji = EMOJI_RE.search(message_id.content)
-        if not emoji:
-            await ctx.send(chat.error(_("No emojis found specified message.")))
-            return
-        url = (
-            "https://cdn.discordapp.com/emojis/"
-            f"{emoji.group(3)}.{'gif' if emoji.group(2) else 'png'}?v=1"
-        )
-        async with self.session.get(url) as r:
-            data = await r.read()
+        if isinstance(location, (discord.Emoji, discord.PartialEmoji)):
+            async with self.session.get(str(location.url)) as r:
+                data = await r.read()
+            if not name:
+                name = location.name
+        elif isinstance(location, discord.Message):
+            emoji = EMOJI_RE.search(location.content)
+            if not emoji:
+                await ctx.send(chat.error(_("No emojis found in the specified message.")))
+                return
+            url = (
+                "https://cdn.discordapp.com/emojis/"
+                f"{emoji.group(3)}.{'gif' if emoji.group(2) else 'png'}?v=1"
+            )
+            async with self.session.get(url) as r:
+                data = await r.read()
+        elif isinstance(location, discord.Member):
+            emoji = None
+            if (
+                location.activity
+                and location.activity.emoji
+                and location.activity.emoji.is_custom_emoji()
+            ):
+                emoji = location.activity.emoji
+            if not emoji:
+                await ctx.send(
+                    chat.error(_("This user does not have a custom emoji in their status."))
+                )
+                return
+            async with self.session.get(str(emoji.url)) as r:
+                data = await r.read()
+            name = emoji.name
```

Comment:
This makes `name` arg useless in case of using command for taking emoji from status
![img](https://moeka.is-a-good-waifu.com/AV2r15U.png)